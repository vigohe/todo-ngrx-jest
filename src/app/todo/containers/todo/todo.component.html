<app-input (add)="addTodo($event)"></app-input>
<app-list [todos]="todos | async" (toggle)="onToggle($event)" (remove)="onRemove($event)"></app-list>
