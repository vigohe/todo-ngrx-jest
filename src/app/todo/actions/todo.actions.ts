import { Action } from '@ngrx/store';
import { Todo } from '../models/todo';

export enum TodoActionTypes {
  AddTodo = '[Todo] ADD',
  RemoveTodo = '[Todo] Remove',
  ToggleTodo = '[Todo] Toggle'
}

export class AddTodo implements Action {
  readonly type = TodoActionTypes.AddTodo;
  constructor(public payload : string) {};
}

export class RemoveTodo implements Action {
  readonly type = TodoActionTypes.RemoveTodo;
  constructor(public payload : Todo) {};
}

export class ToggleTodo implements Action {
  readonly type = TodoActionTypes.ToggleTodo;
  constructor(public payload : Todo) {};
}


export type TodoActions = AddTodo | RemoveTodo | ToggleTodo;
