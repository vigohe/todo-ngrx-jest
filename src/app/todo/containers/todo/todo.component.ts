import { Component, OnInit } from '@angular/core';
import { Store } from '@ngrx/store';
import * as fromStore from '../../reducers/todo.reducer';
import { AddTodo } from '../../actions/todo.actions';

@Component({
  selector: 'app-todo',
  templateUrl: './todo.component.html',
  styleUrls: ['./todo.component.css']
})
export class TodoComponent implements OnInit {

  constructor(private store: Store<fromStore.State>) { }

  ngOnInit() {
    this.store.dispatch(new AddTodo('HOLA MUNDO'));
  }

}
