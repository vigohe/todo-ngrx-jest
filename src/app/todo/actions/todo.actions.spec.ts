import { AddTodo, ToggleTodo, RemoveTodo, TodoActions } from './todo.actions';

describe('AddTodo', () => {
  it('should set payload to a Todo', () => {
    const action = new AddTodo('GO TO WORK!');
    expect(action.type).toMatchSnapshot();
    expect(action.payload).toMatchSnapshot();
  });

});
