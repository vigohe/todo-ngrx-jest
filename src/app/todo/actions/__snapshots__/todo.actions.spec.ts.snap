// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo should set payload to a Todo 1`] = `"[Todo] ADD"`;

exports[`AddTodo should set payload to a Todo 2`] = `"GO TO WORK!"`;
